src/Features/LanguageServer/Protocol/Handler/Completion/CompletionHandler.cs

```diff
@@ -54,7 +54,7 @@ internal class CompletionHandler : IRequestHandler<LSP.CompletionParams, LSP.Com
             // C# and VB share the same LSP language server, and thus share the same default trigger characters.
             // We need to ensure the trigger character is valid in the document's language. For example, the '{'
             // character, while a trigger character in VB, is not a trigger character in C#.
-            var triggerCharacter = char.Parse(request.Context.TriggerCharacter);
+            char.TryParse(request.Context.TriggerCharacter, out var triggerCharacter);
```

Comment:
The good news is that in an updated version of VS LSP library this is correctly annotated and responding to those annotations [found the bug](https://github.com/dotnet/roslyn/pull/47882/files#diff-0735f44f689a499a4e9ecb6aa4a5c2deR54).

Also nit: I think it would be better to explicitly handle null here, and not just silently fail for _any_ bad data, so we can identify future breaks, like someone sending a multi-character string.

Reply:
I definitely prefer the null check here as I wouldn't expect LSP to send us chars we can't parse - so throwing seems reasonable?

Reply:
yes please.

Reply:
If the trigger kind is `TriggerKind.TriggerCharacter` then the `context.TriggerCharacter` can never be `null`; however, if the trigger kind is **not** `TriggerKind.TriggerCharacter` then the trigger character should always be `null`. Just saying because `null` is a verrrry valid option according to the spec and can definitely result in completions, i.e. just typing any identifier would have this behavior. Soo I don't think throwing here makes sense.

That being said @davidwengier you'll have to expand your `null` handling to not prevent all completion in your PR: https://github.com/dotnet/roslyn/pull/47882/files#diff-0735f44f689a499a4e9ecb6aa4a5c2deR54

Reply:
Is this pr superseded by https://github.com/dotnet/roslyn/pull/47882/files#diff-0735f44f689a499a4e9ecb6aa4a5c2deR54 then?

Reply:
Yes